.github/workflows/sdist.yml

```diff
@@ -0,0 +1,59 @@
+name: sdist
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+      - 1.2.x
+      - 1.3.x
+    paths-ignore:
+      - "doc/**"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: sdist test
+    timeout-minutes: 60
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        pip install numpy python-dateutil pytz
+        pip list
+
+    - name: Build Pandas sdist
+      run: |
+        python setup.py sdist --formats=gztar
+
+    - name: Install Pandas From sdist
+      run: |
+        pip install cython hypothesis pytest pytest-xdist
```

Comment:
It should not be needed to install those first, pip should create a build environment that adds the required build dependencies (cython). I would only install pytest afterwards when actually using it to run the tests

Reply:
> It should not be needed to install those first, pip should create a build environment that adds the required build dependencies (cython).

yes the second details box in https://github.com/pandas-dev/pandas/issues/39416#issuecomment-861457087 is the install from sdist inside a clean conda env.

<details>

```
(pandas-dev) simon@P340:~/pandas (master)$ conda create -n build-pandas python=3.9
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.9.2
  latest version: 4.10.1

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/simon/miniconda3/envs/build-pandas

  added / updated specs:
    - python=3.9


The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-4.5-1_gnu
  ca-certificates    pkgs/main/linux-64::ca-certificates-2021.5.25-h06a4308_1
  certifi            pkgs/main/linux-64::certifi-2021.5.30-py39h06a4308_0
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.35.1-h7274673_9
  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.3.0-h5101ec6_17
  libgomp            pkgs/main/linux-64::libgomp-9.3.0-h5101ec6_17
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.3.0-hd4cf53a_17
  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1
  openssl            pkgs/main/linux-64::openssl-1.1.1k-h27cfd23_0
  pip                pkgs/main/linux-64::pip-21.1.2-py39h06a4308_0
  python             pkgs/main/linux-64::python-3.9.5-h12debd9_4
  readline           pkgs/main/linux-64::readline-8.1-h27cfd23_0
  setuptools         pkgs/main/linux-64::setuptools-52.0.0-py39h06a4308_0
  sqlite             pkgs/main/linux-64::sqlite-3.35.4-hdfb4753_0
  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0
  tzdata             pkgs/main/noarch::tzdata-2020f-h52ac0ba_0
  wheel              pkgs/main/noarch::wheel-0.36.2-pyhd3eb1b0_0
  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate build-pandas
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(pandas-dev) simon@P340:~/pandas (master)$ conda activate build-pandas
(build-pandas) simon@P340:~/pandas (master)$ pip install ~/downloads/pandas-1.3.0rc1.tar.gz
Processing /home/simon/downloads/pandas-1.3.0rc1.tar.gz
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting pytz>=2017.3
  Using cached pytz-2021.1-py2.py3-none-any.whl (510 kB)
Collecting numpy>=1.17.3
  Using cached numpy-1.20.3-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.4 MB)
Collecting python-dateutil>=2.7.3
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Building wheels for collected packages: pandas
  Building wheel for pandas (PEP 517) ... done
  Created wheel for pandas: filename=pandas-1.3.0rc1-cp39-cp39-linux_x86_64.whl size=12076754 sha256=91c50656fd1add111e298747815f1065884472f8d9db92e47e45041319d24add
  Stored in directory: /home/simon/.cache/pip/wheels/ab/30/e9/e4606c8789f7fa8438719c11a5515eafa5fcbfbc5107a746c6
Successfully built pandas
Installing collected packages: six, pytz, python-dateutil, numpy, pandas
Successfully installed numpy-1.20.3 pandas-1.3.0rc1 python-dateutil-2.8.1 pytz-2021.1 six-1.16.0
(build-pandas) simon@P340:~/pandas (master)$ python
Python 3.9.5 (default, Jun  4 2021, 12:28:51) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
(build-pandas) simon@P340:~/pandas (master)$ cd ..
(build-pandas) simon@P340:~$ python
Python 3.9.5 (default, Jun  4 2021, 12:28:51) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> pd.__version__
'1.3.0rc1'
>>> 
```
</details>

now that does not fail on the pandas import as setuptools is installed with Python using the `conda create` command, unlike the conda test on pandas-feedstock.




> I would only install pytest afterwards when actually using it to run the tests

again maybe use a clean conda environment to do the testing.

Reply:
> > I would only install pytest afterwards when actually using it to run the tests
> 
> again maybe use a clean conda environment to do the testing.

maybe we don't need to run the tests in this workflow. and use a matrix to just check that the sdist builds (and imports) on the different supported Python versions instead.